solr/core/src/java/org/apache/solr/servlet/CoordinatorHttpSolrCall.java

```diff
@@ -87,22 +88,18 @@ public static SolrCore getCore(
 
         DocCollection syntheticColl = clusterState.getCollectionOrNull(syntheticCollectionName);
         if (syntheticColl == null) {
-          // no such collection. let's create one
+          // no synthetic collection for this config, let's create one
           if (log.isInfoEnabled()) {
             log.info(
                 "synthetic collection: {} does not exist, creating.. ", syntheticCollectionName);
           }
           createColl(syntheticCollectionName, solrCall.cores, confName);
-          // Try to get core again now that we have created the synthetic core
-          core = solrCall.getCoreByCollection(collectionName, isPreferLeader);
+          syntheticColl =
+              zkStateReader.getClusterState().getCollectionOrNull(syntheticCollectionName);
         }
-        if (core != null) {
-          factory.collectionVsCoreNameMapping.put(collectionName, core.getName());
-          solrCall.cores.getZkController().getZkStateReader().registerCore(collectionName);
-          if (log.isDebugEnabled()) {
-            log.debug("coordinator node, returns synthetic core: {}", core.getName());
-          }
-        } else {
+        List<Replica> nodeNameSyntheticReplicas =
```

Comment:
Thanks @justinrsweeney.
I'm just adding couple of notes for followup
1. We may need to add sync to create core locally as with multiple threads it may end up creating more replicas (atleast I don't see any sync)?
2. Its not clear to me that we need to register synthetic collection in ZK - maybe there is some benefit of it?
3. Just wanted to make sure same synthetic-core across collections/orgs not causing any interfere.

Reply:
Sounds good, I think this should get us past the key issues and then we can sync on Tuesday and carve out other next steps.